galsim/utilities.py

```diff
@@ -418,6 +418,119 @@ def _gammafn(x):
+                             and y directions respectively.
+    @param conserve_flux     Should the routine output images that have, on average, same total
+                             pixel values as the input image (True) or should the pixel values
+                             summed over all the images equal the sum of pixel values of the input
+                             image (False)? [default:False]
+    @param suppress_warnings Suppresses the warnings about the pixel scale of the output, if True.
+                             [default:False]
+
+    @returns a list of images and offsets to reconstruct the input image using 'interleaveImages'.
+    """
+
+    if isinstance(N,int):
+        n1,n2 = N,N
+    elif hasattr(N,'__iter__'):
+        if len(N)==2:
+            n1,n2 = N
+        else:
+            raise TypeError("'N' has to be a list or a tuple of two integers")
+        if not (isinstance(n1,int) and  isinstance(n2,int)):
+            raise TypeError("'N' has to be of type int or a list or a tuple of two integers")
+    else:
+        raise TypeError("'N' has to be of type int or a list or a tuple of two integers")
+
+    if not isinstance(image,galsim.Image):
+        raise TypeError("'image' has to be an instance of galsim.Image")
+
+    y_size,x_size = image.array.shape
+    if x_size%n1 or y_size%n2:
+        raise ValueError("The value of 'N' is incompatible with the dimensions of the image to "+
+                         +"be 'deinterleaved'")
+
+    wcs = image.wcs
+    if wcs is not None:
+        if wcs.isPixelScale():
+            scale = wcs.scale
+    else:
+        scale = None
+
+    im_list, offsets = [], []
+    for i in xrange(n1):
+      for j in xrange(n2):
+        # The tricky part - going from array indices to Image coordinates (x,y)
+        # DX[i'] = -(i+0.5)/n+0.5 = -i/n + 0.5*(n-1)/n
+        #    i  = -n DX[i'] + 0.5*(n-1)
+        dx,dy = -(i+0.5)/n1+0.5,-(j+0.5)/n2+0.5
+        offset = galsim.PositionD(dx,dy)
+        img_arr = image.array[j::n2,i::n1].copy()
+        img = galsim.Image(img_arr)
+        if conserve_flux is True:
+            img *= n1*n2
+        im_list.append(img)
+        offsets.append(offset)
+
+    if scale is not None:
+        if n1==n2:
+            for img in im_list:
+                img.wcs = galsim.PixelScale(scale*n1)
+        else:
+            for img in im_list:
+                img.wcs = galsim.JacobianWCS(scale*n1, 0., 0., scale*n2)
+    elif isinstance(wcs,galsim.JacobianWCS): # from say, a previously interleaved Image.
+        dudx, dudy, dvdx, dvdy = wcs.dudx, wcs.dudy, wcs.dvdx, wcs.dvdy
+        if (dudx*n1==dvdy*n2) and (dudy==0.0) and (dvdx==0.0):
+            for img in im_list:
+                img.wcs = galsim.PixelScale(dudx*n1)
+        else:
+            for img in im_list:
+                img.wcs = galsim.JacobianWCS(dudx*n1,dudy*n2,dvdx*n1,dvdy*n2)
```

Comment:
How important is it to look if the WCS is local? I ask this because I had a test case where I give only a pixel scale and then change the origin of the image to have some non-trivial bounds. The WCS of the image becomes an instance of the `OffsetWCS` class which is non-local.